### DISCUSSION TITLE
Monitoring a PVD file for real-time changes in Trame and Paraview

### DISCUSSION THREAD

**Discussion by NimaGhorab** [Kitware/trame#637] [General] [answered]
Greetings everyone.
I have a solver running in the background that generates a PVD file while solving a CFD problem.
As the simulation progresses, new timesteps are added to the PVD file.
I'm using a Trame web server to display the PVD file with the help of the Paraview Python library.
I'm wondering if Trame or Paraview has a built-in feature to automatically detect and display these new timesteps as soon as the solver adds them to the file or should I use the external third-party libraries like Wachdog in Python?
Thank you so much.

**Comment by jourdain** [comment, marked as answer]
`Wachdog` is indeed the way to go. You can see an example using it in trame with asyncio [in the hot reload section](https://kitware.github.io/trame/examples/core/basics.html#hot-reload).
